2000-08-23  Jason Merrill  <jason@redhat.com>

	* typeck.c (build_ptrmemfunc): Save the input pmf.

*** gcc/cp/typeck.c	2000/08/18 09:31:11	1.304
--- gcc/cp/typeck.c	2000/08/24 01:29:15
*************** build_ptrmemfunc (type, pfn, force)
*** 6161,6166 ****
--- 6161,6168 ----
  	}
        else
  	{
+ 	  if (TREE_SIDE_EFFECTS (pfn))
+ 	    pfn = save_expr (pfn);
  	  ndelta = cp_convert (ptrdiff_type_node, 
  			       build_component_ref (pfn, 
  						    delta_identifier, 


